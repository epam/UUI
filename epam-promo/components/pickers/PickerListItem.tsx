import React from 'react';
import css from './PickerListItem.scss';
import { DataRowProps } from '@epam/uui-core';
import { Checkbox, RadioInput } from '../inputs';
import { TextPlaceholder } from '../typography';

export interface PickerListItemProps<TItem, TId> extends DataRowProps<TItem, TId> {
    getName(item: TItem): string;
}

export class PickerListItem<TItem, TId> extends React.Component<PickerListItemProps<TItem, TId>> {
    render() {
        let label: any;

        if (this.props.isLoading) {
            label = <TextPlaceholder wordsCount={2} />;
        } else {
            label = this.props.getName(this.props.value);
        }

        let component: any;

        if (this.props.checkbox) {
            component = (
                <Checkbox
                    {...this.props.checkbox}
                    isDisabled={this.props.isLoading || this.props.checkbox.isDisabled}
                    label={label}
                    value={this.props.isChecked}
                    onValueChange={() => this.props.onCheck(this.props)}
                />
            );
        } else {
            component = (
                <RadioInput
                    label={label}
                    value={this.props.isSelected}
                    isDisabled={this.props.isLoading || !this.props.isSelectable}
                    onValueChange={() => this.props.onSelect(this.props)}
                />
            );
        }

        return <div className={css.row}>{component}</div>;
    }
}
