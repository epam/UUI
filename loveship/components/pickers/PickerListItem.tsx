import React from 'react';
import { DataRowProps } from '@epam/uui-core';
import { RadioInput, Checkbox } from '../inputs';
import { TextPlaceholder } from '../typography';
import css from './PickerListItem.scss';
import { Theme } from '../types';

export interface PickerListItemProps<TItem, TId> extends DataRowProps<TItem, TId> {
    getName(item: TItem): string;
    theme?: Theme;
}

export class PickerListItem<TItem, TId> extends React.Component<PickerListItemProps<TItem, TId>> {
    render() {
        let label: any;

        if (this.props.isLoading) {
            label = <TextPlaceholder wordsCount={2} />;
        } else {
            label = this.props.getName(this.props.value);
        }

        let component: any;

        if (this.props.checkbox) {
            component = (
                <Checkbox
                    {...this.props.checkbox}
                    isDisabled={this.props.isLoading || this.props.checkbox.isDisabled}
                    label={label}
                    value={this.props.isChecked}
                    onValueChange={() => this.props.onCheck(this.props)}
                    theme={this.props.theme}
                />
            );
        } else {
            component = (
                <RadioInput
                    label={label}
                    value={this.props.isSelected}
                    isDisabled={this.props.isLoading || !this.props.isSelectable}
                    onValueChange={() => this.props.onSelect(this.props)}
                    theme={this.props.theme}
                />
            );
        }

        return <div className={css.row}>{component}</div>;
    }
}
