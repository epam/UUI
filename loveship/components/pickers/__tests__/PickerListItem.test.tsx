import React from 'react';
import renderer from 'react-test-renderer';
import { PickerListItem } from '../PickerListItem';

describe('PickerListItem', () => {
    const id = '1';
    const rowKey = '2';
    const index = 3;
    const getName = jest.fn();

    it('should render with default props', () => {
        const tree = renderer.create(<PickerListItem id={id} rowKey={rowKey} index={index} getName={getName} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render with all new props', () => {
        const tree = renderer.create(<PickerListItem id={id} rowKey={rowKey} index={index} getName={getName} theme="light" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
