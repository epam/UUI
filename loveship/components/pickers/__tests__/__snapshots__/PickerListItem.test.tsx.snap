// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerListItem should render with all new props 1`] = `
<div
  className="row"
>
  <label
    className="container uui-disabled root size-18 color-sky theme-light"
  >
    <div
      className="uui-radioinput"
    >
      <input
        disabled={true}
        onChange={[Function]}
        tabIndex={0}
        type="radio"
      />
    </div>
  </label>
</div>
`;

exports[`PickerListItem should render with default props 1`] = `
<div
  className="row"
>
  <label
    className="container uui-disabled root size-18 color-sky theme-light"
  >
    <div
      className="uui-radioinput"
    >
      <input
        disabled={true}
        onChange={[Function]}
        tabIndex={0}
        type="radio"
      />
    </div>
  </label>
</div>
`;
